deliaioanac/assignment2/PrintAncestorsUnitTests.cpp

```diff
@@ -0,0 +1,66 @@
+#include "printAncestors.cpp"
+#include <gtest/gtest.h>
+#include <vector>
+
+namespace {
+    class PrintAncestorsTest : public ::testing::Test {
+        protected:
+            BinaryTree tree;
+
+            virtual void SetUp() {
+                /*
+                 * Constructing the following tree :
+                 * 
+                 *                1
+                 *               / \
+                 *              /   \
+                 *             2     3
+                 *            / \   / \
+                 *           4   5 6   7
+                 *          / \  /
+                 *         8  9 10
+                 */
+                std::vector<int> elements = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10};
+                tree.createTree(elements);
+            }
+
+            virtual void TearDown() {
+                tree.destroyTree(); 
```

Comment:
If you implement a class destructor, you no longer need a `TearDown` function at all :)

Reply:
Done :)